src/server_manager/web_app/digitalocean_cloud.ts

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020 The Outline Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {createDigitalOceanSession} from '../cloud/digitalocean_api';
+import {Cloud} from '../model/cloud';
+
+import {DigitalOceanAccount} from './digitalocean_account';
+import {ShadowboxSettings} from './shadowbox_server';
+
+export class DigitalOceanCloud implements Cloud {
+  private static readonly ACCOUNT_STORAGE_KEY = 'DigitalOceanAccounts';
+
+  private readonly accounts = new Set<string>();
+
+  constructor(
+      private storage: Storage, private shadowboxSettings: ShadowboxSettings,
+      private debugMode: boolean) {
+    const serialized = storage.getItem(DigitalOceanCloud.ACCOUNT_STORAGE_KEY);
+    if (serialized != null) {
+      (JSON.parse(serialized) as string[]).map((account) => this.accounts.add(account));
+    }
+  }
+
+  connectAccount(accessToken: string): DigitalOceanAccount {
+    this.addAccount(accessToken);
+    return this.createDigitalOceanAccount(accessToken);
+  }
+
+  listAccounts(): DigitalOceanAccount[] {
+    return [...this.accounts].map((accessToken) => this.createDigitalOceanAccount(accessToken));
+  }
+
+  private createDigitalOceanAccount(accessToken: string): DigitalOceanAccount {
+    const doApiClient = createDigitalOceanSession(accessToken);
+    return new DigitalOceanAccount(
+        doApiClient, this.shadowboxSettings, this.debugMode, () => this.deleteAccount(accessToken));
+  }
+
+  private deleteAccount(accessToken: string): void {
```

Comment:
Let's name this disconnectAccount to match the product language. Also, let's make this public and remove DOAccount.disconnect();

We should be able to disconnect by account id, not accessToken. The logic would be to iterate over the list checking the ids (which in my suggestion you already got at createDigitalOceanAccount and passed to the constructor)

Reply:
As a side note, at some point we should be able to refresh the token, without changing the account id.